AdyenSession/Session+ActionComponentDelegate.swift

```diff
@@ -10,22 +10,33 @@ import Adyen
 #endif
 import Foundation
 
+/// :nodoc:
 extension Session: ActionComponentDelegate {
     public func didFail(with error: Error, from component: ActionComponent) {
-        // TODO: call back the merchant
+        didFail(with: error, currentComponent: component)
     }
 
     public func didComplete(from component: ActionComponent) {
+        didComplete(currentComponent: component)
+    }
+    
+    internal func didComplete(currentComponent: Component) {
         // TODO: call back the merchant
     }
 
     public func didProvide(_ data: ActionComponentData, from component: ActionComponent) {
-        (component as? PresentableComponent)?.viewController.view.isUserInteractionEnabled = false
+        didProvide(data, currentComponent: component)
+    }
+    
+    internal func didProvide(_ data: ActionComponentData, currentComponent: Component) {
+        (currentComponent as? PresentableComponent)?.viewController.view.isUserInteractionEnabled = false
```

Comment:
Is it a good idea that Sessions mutating component or it's view?
Maybe we can do `willProvide` in delegate instead?

Reply:
Now that I am thinking of it - this could be not such a great idea..
Does it affect DropIn? DropIn is a PresentableComponent, but if it show "AwaitComponent" and this will disable user's interaction..